src/v2/Artsy/Relay/middleware/cache/__tests__/Cache.jest.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * @jest-environment node
+ */
+
+import { Cache } from "../Cache"
+
+jest.mock("lib/environment")
+
+describe("Cache", () => {
+  const getCache = (props = {}) => {
+    return new Cache({
+      size: 10,
+      ttl: 900000,
+      ...props,
+    })
+  }
+
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  describe("enableServerSideCaching", () => {
+    it("disables if disableServerSideCache=false", () => {
+      const cache = getCache({ disableServerSideCache: true })
+      expect(cache.enableServerSideCache).toBe(false)
+    })
+
+    it("enables if disableServerSideCache=true", () => {
+      const cache = getCache({ disableServerSideCache: false })
+      expect(cache.enableServerSideCache).toBe(true)
+    })
+  })
+
+  it("initializes a relay cache", () => {
+    const spy = jest.spyOn(Cache.prototype, "initRelayCache")
+    const cache = getCache()
+    expect(spy).toHaveBeenCalled()
+    expect(cache.relayCache).toBeTruthy()
+  })
+
+  it("returns a cache key", () => {
+    const cache = getCache()
+    expect(cache.getCacheKey("ArtistQuery", { slug: "picasso" })).toEqual(
+      JSON.stringify({
+        queryId: "ArtistQuery",
+        variables: {
+          slug: "picasso",
+        },
+      })
+    )
+  })
+
+  describe("setting and getting cache", () => {
+    describe("client", () => {
+      it("sets / gets the cache by cacheKey", async () => {
+        const cache = getCache()
+        cache.enableServerSideCache = false
+        const queryId = "ArtistQuery"
+        const variables = { slug: "picasso" }
+        const response = { data: { artist: { slug: "picasso" } } }
+        const options = { cacheConfig: { force: false } }
+        cache.set(queryId, variables, response, options)
+        expect(await cache.get(queryId, variables)).toEqual(
+          expect.objectContaining({ data: { artist: { slug: "picasso" } } })
+        )
+      })
+    })
+
+    describe("server", () => {
+      // it("does not set cache if enableServerSideCaching=false", async () => {
```

Comment:
These are good specs, just need to figure out how to mock the new `cacheClient` properly (and keep these specs/assertions).

Wanted to comment out to see if there was other test issues/flakiness in the global suite on CI before revisiting this known one.